app/services/monthly_district_data_service.rb

```diff
@@ -94,18 +109,34 @@ def section_row
       Array.new(month_headers.size - 1, nil),
       "Treatment outcomes of patients under care",
       Array.new(outcome_headers.size - 1, nil),
+      "Days of patient medications",
+      Array.new((medications_dispensation_headers.size * @medication_dispensation_months.size) - 1, nil),
       "Drug availability",
       Array.new(drug_headers.size - 1, nil)
     ].flatten
   end
 
+  def sub_section_row
+    [
+      Array.new(region_headers.size + summary_headers.size + month_headers.size * 2 + outcome_headers.size, nil),
+      medications_dispensation_month_headers.map do |month|
+        if month == medications_dispensation_headers.last
+          month
+        else
+          [month, Array.new(medications_dispensation_headers.size - 1, nil)]
+        end
+      end
+    ].flatten
```

Comment:
We could lift the `if` out of the loop to make the difference between the other elements and the last one easier to understand. Wdyt of something like:

```ruby
medications_dispensation_headers[0...-1].map do |month|
  [month, Array.new(medications_dispensation_headers.size - 1, nil)] 
end.flatten,
medications_dispensation_headers.last,
```